tests/Bitcoin/BitcoinAddressTests.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright © 2017-2021 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Bitcoin/Address.h"
+#include <TrustWalletCore/TWCoinType.h>
+#include "PublicKey.h"
+#include "Bitcoin/Script.h"
+#include "HexCoding.h"
+
+#include <string>
+#include <gtest/gtest.h>
+
+using namespace TW;
+using namespace TW::Bitcoin;
+
+const char* TestPubKey1 = "039d645d2ce630c2a9a6dbe0cbd0a8fcb7b70241cb8a48424f25593290af2494b9";
+const char* TestP2phkAddr1 = "12dNaXQtN5Asn2YFwT1cvciCrJa525fAe4";
+const char* TestP2phkData1 = "0011d91ce1cc681f95583da3f4a6841c174be950c7";
+const char* TestP2shAddr1 = "3PQ5BD39rDikf7YW6pJ9a9tbS3QhvwvzTG";
+const char* TestP2shData1 = "05ee1e69460b59027d9df0a79ca2c92aa382a25fb7";
+
+TEST(BitcoinAddress, P2PKH_CreateFromString) {
+    const auto address = Address(TestP2phkAddr1);
+    EXPECT_EQ(address.string(), TestP2phkAddr1);
+    EXPECT_EQ(hex(address.bytes), TestP2phkData1);
+}
+
+TEST(BitcoinAddress, P2PKH_CreateFromPubkey) {
+    const auto publicKey = PublicKey(parse_hex(TestPubKey1), TWPublicKeyTypeSECP256k1);
+    const auto address = Address(publicKey, TWCoinTypeP2pkhPrefix(TWCoinTypeBitcoin));
+    EXPECT_EQ(address.string(), TestP2phkAddr1);
+    EXPECT_EQ(hex(address.bytes), TestP2phkData1);
+}
+
+TEST(BitcoinAddress, P2PKH_CreateFromData) {
+    const auto address = Address(parse_hex(TestP2phkData1));
+    EXPECT_EQ(address.string(), TestP2phkAddr1);
+    EXPECT_EQ(hex(address.bytes), TestP2phkData1);
+}
+
+TEST(BitcoinAddress, P2SH_CreateFromString) {
+    const auto address = Address(TestP2shAddr1);
+    EXPECT_EQ(address.string(), TestP2shAddr1);
+    EXPECT_EQ(hex(address.bytes), TestP2shData1);
+}
+
+TEST(BitcoinAddress, P2SH_CreateCreateFromPubkeyScript) {
+    // P2SH address cannot be created directly from pubkey, script is built
+    const auto publicKey = PublicKey(parse_hex(TestPubKey1), TWPublicKeyTypeSECP256k1);
+    const auto pubKeyHash = publicKey.hash({});
```

Comment:
```suggestion

    const auto pubKeyHash = publicKey.hash({});
```
add blank lines before each `EXPECT_EQ` for better readability